models/agent_helpers.go

```diff
@@ -637,6 +637,63 @@ type CreateAgentParams struct {
 	DisableCollectors              []string
 }
 
+func verifyNodeAndAgent(nodeType NodeType, agentType AgentType) error {
+	switch nodeType {
+	case GenericNodeType:
+		return nil
+	case ContainerNodeType:
+		return nil
+	case RemoteNodeType:
+		if agentType == ExternalExporterType {
+			return nil
+		}
+	case RemoteRDSNodeType:
+		if agentType == RDSExporterType {
+			return nil
+		}
+	case RemoteAzureDatabaseNodeType:
+		if agentType == AzureDatabaseExporterType {
+			return nil
+		}
+	default:
+		return nil
+	}
+	return errors.Errorf("invalid combination of node type %s and agent type %s", nodeType, agentType)
+}
+
+func verifyServiceAndAgent(serviceType ServiceType, agentType AgentType) error {
+	switch agentType {
+	case PMMAgentType:
+	case NodeExporterType:
+	case RDSExporterType:
+	case AzureDatabaseExporterType:
+	case VMAgentType:
```

Comment:
I think we can remove these cases and use map mentioned by @oter, because we shouldn't try to bind service for these agents.
For all cases out of that map we can return error.

Reply:
Right now if it's out of map, I return as valid. Should I switch to false? Not sure what is better approach.
if we add some new agent type, allow it by default or forbid it by default.